qa/tasks/ceph_salt.py

```diff
@@ -0,0 +1,274 @@
+        self.nodes_storage_only = self.ctx['nodes_storage_only']
+        self.remotes = self.ctx['remotes']
+        self.roles = self.ctx['roles']
+        self.sm = ceph_salt_ctx['salt_manager_instance']
+        self.role_types = self.ctx['role_types']
+        self.remote_lookup_table = self.ctx['remote_lookup_table']
+        self.ceph_salt_deploy = ceph_salt_ctx['deploy']
+        self.scripts = Scripts(self.ctx, self.log)
+
+    def _install_ceph_salt(self):
+        '''
+        Installs ceph-salt on master either from source if repo and/or branch are 
+        provided in the suite yaml or from rpm if not
+        '''
+        global ceph_salt_ctx
+        if ceph_salt_ctx['repo']:
+            if not ceph_salt_ctx['branch']:
+                self.scripts.run(
+                self.master_remote,
+                'install_ceph_salt.sh',
+                args=ceph_salt_ctx['repo']
+                )
+            else:
+                self.scripts.run(
+                self.master_remote,
+                'install_ceph_salt.sh',
+                args=[ceph_salt_ctx['repo'], ceph_salt_ctx['branch']]
+                )
+        else:
+            self.scripts.run(
+            self.master_remote,
+            'install_ceph_salt.sh'
+            )
+        self.ctx.cluster.run(args=[ 'sudo', 'sh', '-c',
+                                    'systemctl restart salt-minion'])
+        self.master_remote.sh("sudo systemctl restart salt-master")
+
+            
+    def _populate_ceph_salt_context(self):
+        global ceph_salt_ctx
+        ceph_salt_ctx['log_anchor'] = self.config.get('log_anchor', self.log_anchor_str)
+        if not isinstance(ceph_salt_ctx['log_anchor'], str):
+            self.log.warning(
+                "log_anchor was set to non-string value ->{}<-, "
+                "changing to empty string"
+                .format(ceph_salt_ctx['log_anchor'])
+                )
+            ceph_salt_ctx['log_anchor'] = ''
+        ceph_salt_ctx['deploy'] = self.config.get('deploy', True)
+        ceph_salt_ctx['quiet_salt'] = self.config.get('quiet_salt', True)
+        ceph_salt_ctx['salt_manager_instance'] = SaltManager(self.ctx)
+        ceph_salt_ctx['master_remote'] = (
+                ceph_salt_ctx['salt_manager_instance'].master_remote
+                )
+        ceph_salt_ctx['repo'] = self.config.get('repo', None)
+        ceph_salt_ctx['branch'] = self.config.get('branch', None)
+
+    def _add_CA_repo(self):
+        '''
+        Adding `SUSE Internal CA Certificate` that's needed to install the container image
+        '''
+        self.ctx.cluster.run(args=[ 'sudo', 'sh', '-c', 
+                                    'sudo zypper -n addrepo --refresh --no-gpgcheck'
+                                    ' http://download.suse.de/ibs/SUSE:/CA/SLE_15_SP2/ '
+                                    '\'SUSE Internal CA Certificate\'; '
+                                    'zypper -n in ca-certificates-suse'])
```

Comment:
why do you need 'sudo' two times?

Reply:
for improving readability and maintainability can this code be refactored like:
```
        self.ctx.cluster.run(args="""
            sudo zypper -n addrepo --refresh --no-gpgcheck http://download.suse.de/ibs/SUSE:/CA/SLE_15_SP2/ 'SUSE Internal CA Certificate'
            sudo zypper -n in ca-certificates-suse""")
```
or
```
        repo_url = 'http://download.suse.de/ibs/SUSE:/CA/SLE_15_SP2/'
        repo_name = "SUSE Internal CA Certificate"
        self.ctx.cluster.run(args="
            sudo zypper -n addrepo --refresh --no-gpgcheck {url} '{name}'
            sudo zypper -n in ca-certificates-suse".format(url=repo_url, name=repo_name))
```
or something else which will be easier to read.